### PR TITLE
fix as manager and from queryset self types (#1788)

### LINKED ISSUE #1788: As_manager: Self not taken into account
# Bug report
```
from django.db import models
from typing_extensions import Self
class MyQuerySet(BaseQuerySet):
       def example_simple(self) -> Self: ...

class MyModel(models.Model):
     objects = MyQuerySet.as_manager()
reveal_type(MyModel.objects.example_simple())  # -> Revealed type is "<nothing>" 
````

## What's wrong

The issue lies in the type revelation within the custom QuerySet method that is supposed to return Self. Currently, it reveals the type as `<nothing>`, which indicates that the Self type isn't properly recognized or handled.

## How is that should be

The correct and expected behavior should reveal the type accurately, as per the following expectation:
```
reveal_type(MyModel.objects.example_simple())  # -> Revealed type is "myapp.models.ManagerFromMyQuerySet[myapp.models.MyModel]"
```

## System information

- OS:
- `python` 3.12
- `django` 3.2.12
- `mypy` 1.6.1
- `django-stubs` 4.2.5
- `django-stubs-ext` 4.2.5

### PR DESCRIPTION
**Pull request by moranabadie**
# I have made things!

I replace self_type by the manager itself

## Related issues

It fixes : https://github.com/typeddjango/django-stubs/issues/1788

### CODE REVIEW

**Review by sobolevn — APPROVED** [review]
Thanks!

File: tests/typecheck/managers/querysets/test_as_manager.yml [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+        - myapp
+    files:
+        -   path: myapp/__init__.py
+        -   path: myapp/models.py
+            content: |
+                from django.db import models
+                from typing_extensions import Self
+
+                class BaseQuerySet(models.QuerySet):
+                    def example_dict(self) -> dict[str, Self]: ...
```
Comment (sobolevn) [inline tests/typecheck/managers/querysets/test_as_manager.yml]:
```suggestion
                    def example_dict(self) -> Dict[str, Self]: ...
```

File: tests/typecheck/managers/querysets/test_as_manager.yml [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+
+                class MyQuerySet(BaseQuerySet):
+                    def example_simple(self) -> Self: ...
+                    def example_list(self) -> list[Self]: ...
```
Comment (sobolevn) [inline tests/typecheck/managers/querysets/test_as_manager.yml]:
```suggestion
                    def example_list(self) -> List[Self]: ...
```

**Review by sobolevn — COMMENTED** [review]
To fix the CI

File: tests/typecheck/managers/querysets/test_as_manager.yml (hunk 1 above)
Comment (moranabadie) [reply tests/typecheck/managers/querysets/test_as_manager.yml]: oh thank you

File: tests/typecheck/managers/querysets/test_as_manager.yml (hunk 2 above)
Comment (moranabadie) [reply tests/typecheck/managers/querysets/test_as_manager.yml]: thx, fixed

**Review by sobolevn — APPROVED** [review]

**Pull request merged**